lib/compress/zstdmt_compress.c

```diff
@@ -1417,7 +1419,10 @@ size_t ZSTDMT_initCStream_internal(
     if (params.jobSize != 0 && params.jobSize < ZSTDMT_JOBSIZE_MIN) params.jobSize = ZSTDMT_JOBSIZE_MIN;
     if (params.jobSize > (size_t)ZSTDMT_JOBSIZE_MAX) params.jobSize = (size_t)ZSTDMT_JOBSIZE_MAX;
 
+/* always trigger mt when fuzzing */
+#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
```

Comment:
If it was fixed when this ifdef was added, does that mean that there is a bug in `singleBlockingThread` mode then?

We should either find and fix that bug, or delete single blocking thread mode.

Reply:
Hmm maybe, what is `singleBlockingThread` mode for again? For long mode to run before the other compression jobs?

Reply:
No, `singleBlockingThread` is "single-threaded" mode inside of zstdmt. It is the equivalent of switching to single threaded for inputs `< ZSTDMT_JOBSIZE_MIN`.

Originally zstdmt was its own API, and handled automatic transition back to single threaded mode. Now it is wrapped by the zstd advanced API, and the zstd streaming API handles that transition.

That means `singleBlockingThread` is mostly dead code. The only use case is if people are using the `ZSTDMT_*` API directly. `python-zstandard` used to use this API, but they've switched away. They were the only user I was aware of, so maybe we can remove this functionality.

I'd say lets try to fix it, and see what @Cyan4973 thinks about deleting it.

Reply:
I'll remove it in a different pr